src/garage/tf/models/categorical_gru_model.py

```diff
@@ -60,7 +60,7 @@ def __init__(self,
                  hidden_b_init=tf.zeros_initializer(),
                  recurrent_nonlinearity=tf.nn.sigmoid,
                  recurrent_w_init=tf.initializers.glorot_uniform(),
-                 output_nonlinearity=None,
```

Comment:
if this is necessary for the correct operation of the model, it should not be an option.

Alternatively, you could rename this something like `output_normalization_fn`, but I recommend just removing it.

Reply:
Right, we should have enforced it for categorical primitives. Will fix.

Reply:
there's also no reason you can't have an output non-linearity followed by your normalizer. that seems like another way out of this problem

Reply:
Indeed, how about applying softmax inside the model, while making `output_nonlineaity=None` by default? Then we can still have any output non-linearity as we want.

Reply:
lgtm